activestorage/storage.py

```diff
@@ -8,6 +8,8 @@ def reduce_chunk(rfile, offset, size, compression, filters, missing, dtype, shap
     
     rfile - the actual file with the data 
     offset, size - where and what we want ...
+    compression - optional `numcodecs.abc.Codec` compression codec
```

Comment:
Do we really want to pass an actual codec instance, I'd rather have a complete separation of concerns and pass an identifier for a particular codec, just as is done in the NetCDF API ...  after all this is our API layer???

Reply:
It's a fair point. I was quite surprised to see an actual codec instance being passed through, although it did make the implementation trivial.

Would it make sense to align the S3 active storage API parameters and the storage driver parameters used here? 

Currently I'm using a lower case string to describe the (de)compression algorithm. I need to check that that no parameters are required for other decompression using other algorithms. Filters may need parameters. Shuffle needs to know the data type size, although we do already have that info elsewhere.

Reply:
It'd be good to know how other communities are handling this. How does Zarr doit? We should find out.

Reply:
In terms of the Zarr Python API, it looks like numcodecs is the interface: https://zarr.readthedocs.io/en/stable/api/codecs.html

I believe it was originally part of zarr, then got extracted into a separate library.

Reply:
OK, that's not so helpful I guess as they have the entire block in memory client-side so they can put it all in the stack. Our problem of course is that the client and server are not on the same stack, so we have to go through an API interface. Where we put that is of course a moot point.

Reply:
I'd say that the main API to get right is that of the active storage server. It's JSON based, so can't accept a Python object, but in my current implementation I'm just grabbing the `codec_id` from the numcodecs Codec object, which happens to match my API.

If you want this reduce_chunk interface to form a driver API for this library then it would also be important to keep it simple and stable, although currently all "drivers" are in-tree in this repo.

Is this something we should iterate on from here, with an issue in the backlog?

Reply:
https://github.com/valeriupredoi/PyActiveStorage/issues/125